packages/editor/src/components/entities-saved-states/index.js

```diff
@@ -148,17 +154,31 @@ export default function EntitiesSavedStates( { close } ) {
 					editEntityRecord( kind, name, key, { status: 'publish' } );
 				}
 
-				saveEditedEntityRecord( kind, name, key );
+				pendingSavedRecords.push(
+					saveEditedEntityRecord( kind, name, key )
+				);
 			}
 		} );
 		if ( siteItemsToSave.length ) {
-			saveSpecifiedEntityEdits(
-				'root',
-				'site',
-				undefined,
-				siteItemsToSave
+			pendingSavedRecords.push(
+				saveSpecifiedEntityEdits(
+					'root',
+					'site',
+					undefined,
+					siteItemsToSave
+				)
 			);
 		}
+
+		Promise.all( pendingSavedRecords ).then( ( values ) => {
```

Comment:
Could `pendingSavedRecords` fail? Might be easy enough to also add `catch` to handle it with error notices too.

Reply:
`pendingSavedRecords` is an array of the results of calling `saveEntityRecord`, which afaik never returns an error, just a record or `undefined`. That's why I'm checking for 'undefined' when showing the failure notice. Would it still be worth adding a `catch` in this case? We wouldn't really be doing anything with the caught error.

Reply:
That's a bit unexpected TBH, why wouldn't `saveEntityRecord` throw an error if it... errors? I guess adding a `catch` case does no harm either, in case we want to revisit the behavior of `saveEntityRecord`.

Reply:
I'm not sure what the history behind this is, but the savePost function accounts for it by fetching errors in a [super roundabout way](https://github.com/WordPress/gutenberg/blob/trunk/packages/editor/src/store/actions.js#L229). I'm guessing any changes to `saveEntityRecord` at this point would require a fair bit of refactoring anyway 😅 

Not to say we can't add in a `catch` here, it won't hurt to do so. But we'll still have to generate the failure notices based on `undefined` as the return value too.